src/constructs/dns/dns-records.test.ts

```diff
@@ -15,27 +17,30 @@ describe("The GuDnsRecordSet construct", () => {
     });
     expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
   });
+  it("should use the exact logical id that is passed in", () => {
```

Comment:
We were already getting this behaviour thanks to `CfnResource`, but I have added this test to make it more explicit and ensure that we don't accidentally drop this requirement in the future.

Reply:
Brilliant!